### PR TITLE
test/e2e/apimachinery/watchlist: select only well-known secrets

### PR DESCRIPTION
**Pull request by p0lyn0mial**
the test should list only well-known secrets, as some platforms might add additional secrets specific to their namespace.

#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/apimachinery/watchlist.go [hunk 1]
```diff
@@ -325,6 +348,11 @@ func addWellKnownUnstructuredSecrets(ctx context.Context, f *framework.Framework
 	return secrets
 }
 
+func withWellKnownListOptions(options metav1.ListOptions) metav1.ListOptions {
+	options.LabelSelector = "watchlist=true"
```
Comment (sttts) [inline test/e2e/apimachinery/watchlist.go:352]: would rather make this explicit in the test and not hide it behind a helper.

File: test/e2e/apimachinery/watchlist.go [hunk 2]
```diff
@@ -274,28 +277,48 @@ func verifyStore(ctx context.Context, expectedSecrets []v1.Secret, store cache.S
 }
 
 // corresponds to a streaming request made by the client to stream the secrets
-const expectedStreamingRequestMadeByClient string = "allowWatchBookmarks=true&resourceVersionMatch=NotOlderThan&sendInitialEvents=true&watch=true"
+func getExpectedStreamingRequestMadeByClient() string {
```
Comment (sttts) [inline test/e2e/apimachinery/watchlist.go:280]:
```
var expectedStreamingRequestMadeByClient = func() string {
  ...
} ()
```

no need for a function. Init func would another option.

**Comment by sttts** [comment]
Some nits.

File: test/e2e/apimachinery/watchlist.go [hunk 3]
```diff
@@ -53,12 +54,14 @@ var _ = SIGDescribe("API Streaming (aka. WatchList)", framework.WithSerial(), fu
...
 		stopCh := make(chan struct{})
 		defer close(stopCh)
+
 		secretInformer := cache.NewSharedIndexInformer(
 			&cache.ListWatch{
 				ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
 					return nil, fmt.Errorf("unexpected list call")
 				},
 				WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+					options = withWellKnownListOptions(options)
```
Comment (aaron-prindle) [inline test/e2e/apimachinery/watchlist.go:64]: nit: I believe you can remove this line and have the fxn in the params, eg:
```golang
return f.ClientSet.CoreV1().Secrets(f.Namespace.Name).Watch(context.TODO(), withWellKnownListOptions(options))
```
+1 to what @sttts mentioned though, as it is likely more readable to make this explicit via inlining the options (vs the helper)

Comment (sttts) [reply test/e2e/apimachinery/watchlist.go:64]: Another 2 cent: `withWellKnownListOptions`  is well known now to us, not to any ready in a few month. If we want a helper, call it `withWatchListLabel` or so. Then it is clear without looking it up what this does.

**Comment by p0lyn0mial** [comment]
addressed the comments, thanks for the review, ptal.

File: test/e2e/apimachinery/watchlist.go [hunk 4]
```diff
@@ -274,28 +277,48 @@ func verifyStore(ctx context.Context, expectedSecrets []v1.Secret, store cache.S
...
+func getExpectedRequestsMadeByClientWhenFallbackToListFor(rv string) []string {
 	expectedRequestMadeByClient := []string{
 		expectedStreamingRequestMadeByClient,
 		// corresponds to a list request made by the client
-		"",
+		func() string {
+			params := url.Values{}
+			params.Add("labelSelector", "watchlist=true")
+			return params.Encode()
+		}(),
```
Comment (sttts) [inline test/e2e/apimachinery/watchlist.go:317]: nit: this one looks like it could be a simple string

**Comment by sttts** [comment]
Thanks for the changes.

/lgtm
/approve

**Pull request merged**